### PR TITLE
add SRO pre-release

### PR DESCRIPTION
**Pull request by novacain1**
adds an implementation of SRO in 4.8 for driver related work

### CODE REVIEW

File: base/operators/openshift-special-resource-operator/01-install.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+kind: Namespace
+metadata:
+ name: openshift-special-resource-operator
+---
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+ name: special-resource-operators
+ namespace: openshift-special-resource-operator
+spec:
```
Comment (williamcaban) [inline base/operators/openshift-special-resource-operator/01-install.yaml:12]: Should this be` spec: {} ` ? When empty it should have something indicating that is the case.

File: base/operators/openshift-special-resource-operator/02-node-feature-discovery.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
+---
+apiVersion: nfd.openshift.io/v1
+kind: NodeFeatureDiscovery
```
Comment (williamcaban) [inline base/operators/openshift-special-resource-operator/02-node-feature-discovery.yaml:3]: - This has dependency on the Node Feature Discovery (NFD) Operator. Can we include a README.md with that pointer?

**Review by williamcaban — CHANGES_REQUESTED** [review]

File: base/operators/openshift-special-resource-operator/02-node-feature-discovery.yaml (hunk 2 above)
Comment (novacain1) [reply base/operators/openshift-special-resource-operator/02-node-feature-discovery.yaml:3]: As of this writing, NFD is installed by SRO.  Let's add a separate PR to add NFD if someone wants to add it stand-alone.

Comment (williamcaban) [reply base/operators/openshift-special-resource-operator/02-node-feature-discovery.yaml:3]: Okay. Thanks for the clarification here. Wondering how this is handled if NFD is already installed in the cluster. I'll put this one as an issue for testing

File: base/operators/openshift-special-resource-operator/01-install.yaml (hunk 1 above)
Comment (novacain1) [reply base/operators/openshift-special-resource-operator/01-install.yaml]: updated

**Review by williamcaban — APPROVED** [review]
/LGTM

**Pull request merged**